bentoml/_internal/frameworks/picklable.py

```diff
@@ -1,112 +1,136 @@
+except ImportError:  # pragma: no cover
+    raise MissingDependencyException(
+        f"""cloudpickle is required in order to use the module `{MODULE_NAME}`, install
+         cloudpickle with `pip install cloudpickle`.
+        """
+    )
 
-    Args:
-        tag (:code:`Union[str, Tag]`):
-            Tag of a saved model in BentoML local modelstore.
-        model_store (:mod:`~bentoml._internal.models.store.ModelStore`, default to :mod:`BentoMLContainer.model_store`):
-            BentoML modelstore, provided by DI Container.
 
-    Returns:
-        :obj: an instance of :obj: model from BentoML modelstore.
+logger = logging.getLogger(__name__)
 
-    Examples:
 
-    .. code-block:: python
+def get(tag_like: str | Tag) -> Model:
+    model = bentoml.models.get(tag_like)
+    if model.info.module not in (MODULE_NAME, __name__):
+        raise NotFound(
+            f"Model {model.tag} was saved with module {model.info.module}, failed loading with {MODULE_NAME}."
+        )
+    return model
 
-        import bentoml
 
-        unpickled_model = bentoml.picklable_model.load('my_model:latest')
+def load_model(bento_model: str | Tag | Model) -> ModelType:
     """
-    model = model_store.get(tag)
+    Load the scikit-learn model with the given tag from the local BentoML model store.
 
-    if model.info.module not in (MODULE_NAME, __name__):
+    Args:
+        bento_model (``str`` ``|`` :obj:`~bentoml.Tag` ``|`` :obj:`~bentoml.Model`):
+            Either the tag of the model to get from the store, or a BentoML `~bentoml.Model`
+            instance to load the model from.
+        ...
+    Returns:
+        ``BaseEstimator`` ``|`` ``Pipeline``:
+            The scikit-learn model loaded from the model store or BentoML :obj:`~bentoml.Model`.
+    Example:
+    .. code-block:: python
+        import bentoml
+
+        picklable_model = bentoml.picklable_model.load_model('my_model:latest')
+    """  # noqa
+    if not isinstance(bento_model, Model):
+        bento_model = get(bento_model)
+
+    if bento_model.info.module not in (MODULE_NAME, __name__):
         raise BentoMLException(
-            f"Model {tag} was saved with module {model.info.module}, cannot load with {MODULE_NAME}."
+            f"Model {bento_model.tag} was saved with module {bento_model.info.module}, failed loading with {MODULE_NAME}."
         )
+    model_file = bento_model.path_of(f"{SAVE_NAMESPACE}{PKL_EXT}")
 
-    model_file = model.path_of(f"{SAVE_NAMESPACE}{PKL_EXT}")
     with open(model_file, "rb") as f:
         # The protocol version used is detected automatically, so we do not
         # have to specify it.
         return cloudpickle.load(f)
 
-
 def save_model(
     name: str,
-    obj: t.Any,
+    model: ModelType,
     *,
-    labels: t.Optional[t.Dict[str, str]] = None,
-    custom_objects: t.Optional[t.Dict[str, t.Any]] = None,
-    metadata: t.Optional[t.Dict[str, t.Any]] = None,
+    signatures: dict[str, ModelSignature] | None = None,
```

Comment:
since this is user facing API, could we make it support `dict[str, t.Any]` as well?